### DISCUSSION TITLE
Self clustering images in timeline

### DISCUSSION THREAD

**Discussion by sirebellum** [immich-app/immich#6287] [General]
Hi there,

Some times, especially during burst shots, images look very similar if not identical. I could see a K-means algorithm or something doing some clustering on the images and assigning time-sensitive bins to them. You could stack them automatically!

Let me know your thoughts!

**Comment by mertalev** [comment]
As it happens, we're working on an auto-stacking feature right now! The general idea is to group by a combination of image similarity (possibly with CLIP embeddings) and EXIF metadata (similar time, same camera, etc.).

The approach for the similarity part is still in the air pending testing, but I think a kNN search on each asset with a similarity threshold would work well. K-means could work for simpler cases, but I think it'd be tricky to know the right K in advance.

**Reply by flightmansam** [reply]
That's fantastic!!!

**Reply by VelocityRa** [reply]
Hi, any news on this @mertalev?

**Reply by mertalev** [reply]
The first pass of this was for the duplicate detection utility. We elected not to make it auto-stack to let the user manage them how they'd like. It'd be good to have the option to make it auto-stack, though.

**Reply by VelocityRa** [reply]
Ahh I see. Yeah I'll set up a dev env and look into this when I get some time.

**Comment by MWP** [comment]
Yes please!
Auto-stack by time would be brilliant.
For example: auto stack photos (on import, or by manual job) that are taken within 30sec of each other.
